### ISSUE TITLE
com.google.common.base not exported in MANIFEST.MF

### ISSUE THREAD

**Issue by neadamthal** [google/guava#2155]
I'm new to osgi so perhaps I'm missing the obvious, but I have code that uses some classes in com.google.common.base, but because it is not exported in the MANIFEST.MF file, I am getting resolution errors.

I am working w/ version 15, but took at a look at v18 as well, and it looks like this package is still not exported.

Can someone tell me why that package isn't exported?  or if they have an inkling of what I"m doing wrong?

**Comment by cgdecker** [comment]
I'm looking at the MANIFEST.MF file for Guava 18, and it looks like it's exported there to me. I've highlighted it below with > and **.

```
Manifest-Version: 1.0
Bnd-LastModified: 1408992499326
Build-Jdk: 1.7.0-google-v6
Built-By: cgdecker
Bundle-Description: Guava is a suite of core and expanded libraries that
  include    utility classes, google's collections, io classes, and much
     much more.    Guava has only one code dependency - javax.annotation
 ,    per the JSR-305 spec.
Bundle-DocURL: https://guava-libraries.googlecode.com/
Bundle-License: http://www.apache.org/licenses/LICENSE-2.0.txt
Bundle-ManifestVersion: 2
Bundle-Name: Guava: Google Core Libraries for Java
Bundle-SymbolicName: com.google.guava
Bundle-Version: 18.0.0
Created-By: Apache Maven Bundle Plugin
Export-Package: com.google.common.net;uses:="javax.annotation,com.google
 .common.base,com.google.common.hash,com.google.common.io,com.google.com
 mon.primitives,com.google.common.collect,com.google.common.escape";vers
 ion="18.0.0",com.google.common.html;uses:="com.google.common.escape,jav
 ax.annotation";version="18.0.0",com.google.common.collect;uses:="com.go
 ogle.common.base,javax.annotation,com.google.common.primitives,com.goog
 le.common.math";version="18.0.0",com.google.common.primitives;uses:="co
> m.google.common.base,javax.annotation,sun.misc";version="18.0.0",**com.go
> ogle.common.base;uses:="javax.annotation";version="18.0.0"**,com.google.c
 ommon.escape;uses:="com.google.common.base,javax.annotation";version="1
 8.0.0",com.google.common.cache;uses:="com.google.common.collect,com.goo
 gle.common.util.concurrent,javax.annotation,com.google.common.base,com.
 google.common.primitives,sun.misc";version="18.0.0",com.google.common.e
 ventbus;uses:="com.google.common.collect,com.google.common.cache,javax.
 annotation,com.google.common.base,com.google.common.util.concurrent,com
 .google.common.reflect";version="18.0.0",com.google.common.util.concurr
 ent;uses:="com.google.common.base,javax.annotation,com.google.common.co
 llect,com.google.common.primitives,com.google.common.math";version="18.
 0.0",com.google.common.hash;uses:="com.google.common.primitives,com.goo
 gle.common.base,javax.annotation,com.google.common.math";version="18.0.
 0",com.google.common.io;uses:="javax.annotation,com.google.common.base,
 com.google.common.math,com.google.common.hash,com.google.common.collect
 ,com.google.common.primitives";version="18.0.0",com.google.common.xml;u
 ses:="com.google.common.escape,javax.annotation";version="18.0.0",com.g
 oogle.common.reflect;uses:="javax.annotation,com.google.common.base,com
 .google.common.collect,com.google.common.primitives";version="18.0.0",c
 om.google.common.math;uses:="com.google.common.base,com.google.common.p
 rimitives,javax.annotation";version="18.0.0",com.google.common.annotati
 ons;version="18.0.0"
Import-Package: javax.annotation;resolution:=optional,sun.misc;resolutio
 n:=optional
Tool: Bnd-1.50.0
```

**Comment by neadamthal** [comment]
apologies - my eyes must have been crossed.

On Fri, Sep 18, 2015 at 5:39 PM, Colin Decker <notifications@github.com>
wrote:

> Closed #2155 <https://github.com/google/guava/issues/2155>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/guava/issues/2155#event-413619511>.
>



-- 
Adam Morgan
Salesforce ExactTarget Marketing Cloud <http://www.exacttarget.com/>
SMTS, Technical Services
